src/app/withdraw/page.tsx

```diff
@@ -48,26 +48,39 @@ const Withdraw: React.FC = () => {
 
     const widthDrawFunc = async () => {
         if (!toAddress || !address || !amount) return
-        const amountInShannon = BI.from(amount).mul(BI.from(10).pow(8));
+        const amountInShannon = BI.from(parseFloat(amount) * 10 ** 8)
         let txSkeleton = helpers.TransactionSkeleton({ cellProvider: indexer });
         const collectedCells: Cell[] = [];
         let collector = indexer.collector({lock: helpers.parseAddress(address), type: 'empty'});
         let collectedSum = BI.from(0);
+
         for await (const cell of collector.collect()) {
             collectedSum = collectedSum.add(cell.cellOutput.capacity);
             collectedCells.push(cell);
             if (BI.from(collectedSum).gte(amountInShannon)) break;
         }
-        const transferOutput: Cell = {
+
+        const outputCell: Cell[] = [{
             cellOutput: {
                 capacity: BI.from(amountInShannon).toHexString(),
                 lock: helpers.parseAddress(toAddress),
             },
             data: "0x",
-        };
+        }];
+
+        if (balance > parseFloat(amount)) {
```

Comment:
Think using the difference between Input’s capacity and the output‘s capacity for the judgment instead. 
For instance, `collectedSum.sub(amountInShannon).toNumber() > 0`

Ensure that the judgment condition matches the inputs/outputs of the transaction, rather than calculating with “parseFloat(amount)” separately.

Reply:
Yes, it would be more appropriate to use shannon for the judgment directly. Here, the previous balance judgment has been replaced with shannon.